packages/webapp/src/containers/Animals/AddAnimals/AddAnimalDetails/index.tsx

```diff
@@ -13,8 +13,186 @@
  *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
  */
 
+import { FieldArrayWithId, useFieldArray, useFormContext } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import styles from './styles.module.scss';
+import { Info, Semibold } from '../../../../components/Typography';
+import ExpandableItem from '../../../../components/Expandable/ExpandableItem';
+import useExpandable from '../../../../components/Expandable/useExpandableItem';
+import { AnimalFormHeaderItem } from '../../../../components/Animals/AddAnimalsForm/AnimalFormHeaderItem';
+import AnimalDetails from '../../../../components/Animals/AddAnimalsDetails';
+import BatchDetails from '../../../../components/Animals/AddBatchDetails';
+import { useCurrencySymbol } from '../../../hooks/useCurrencySymbol';
+import { useAnimalOptions } from '../useAnimalOptions';
+import { STEPS } from '..';
+import { DetailsFields } from '../types';
+import { AddAnimalsFormFields } from '../types';
+import { AnimalOrBatchKeys } from '../../types';
+import { parseUniqueDefaultId } from '../../../../util/animal';
+import { getDefaultAnimalIconName } from '../../Inventory/useAnimalInventory';
+import usePopulateDetails from './usePopulateDetails';
+
+type AnimalDetailsField = FieldArrayWithId<AddAnimalsFormFields, 'details'>;
+
 const AddAnimalDetails = () => {
-  return <div>Add animal details</div>;
+  const { expandedIds, toggleExpanded } = useExpandable({ isSingleExpandable: true });
+
+  const { control, watch } = useFormContext<AddAnimalsFormFields>();
+
+  const { fields, remove, replace } = useFieldArray({
+    name: STEPS.DETAILS,
+    control,
+  });
+
+  const onRemoveCard = (index: number): void => {
+    remove(index);
+  };
+
+  const {
+    defaultTypes,
+    sexOptions,
+    sexDetailsOptions,
+    useOptions,
+    tagTypeOptions,
+    tagColorOptions,
+    organicStatusOptions,
+    originOptions,
+  } = useAnimalOptions(
+    'default_types',
+    'sex',
+    'sexDetails',
+    'use',
+    'tagType',
+    'tagColor',
+    'organicStatus',
+    'origin',
+  );
+
+  /* Populate details form based on basics data */
+  usePopulateDetails(replace);
+
+  /* Render logic */
+  const generalDetailProps = {
+    sexOptions,
+    sexDetailsOptions,
+  };
+
+  const otherDetailsProps = {
+    organicStatusOptions,
+  };
+
+  const currency = useCurrencySymbol();
+  const originProps = {
+    currency: currency,
+    originOptions,
+  };
```

Comment:
Now that you created commonProps, I feel like these could directly go into commonProps. You can leave it as is, I'm just obsessed with reducing lines 😂

Reply:
I wrote it that way too at first! But then I wondered if it was more efficient to define the props that don't change outside of the mapping callback function, so they are created only once? But I really wasn't sure -- it's probably a negligible difference, right??! I'll move them because I prefer not having to click through another definition to see what's in the prop 🙂